src/nvim/ex_cmds2.c

```diff
@@ -3403,16 +3413,20 @@ static void script_host_execute_file(char *name, exarg_T *eap)
   // filename
   list_append_string(args, buffer, -1);
   // current range
-  list_append_number(args, eap->line1);
-  list_append_number(args, eap->line2);
+  assert(eap->line1 <= INT_MAX);
+  assert(eap->line2 <= INT_MAX);
+  list_append_number(args, (int)eap->line1);
+  list_append_number(args, (int)eap->line2);
```

Comment:
Because a linenumber is guaranteed to fit in an `int` (yes, one day we might want to lift the restriction and allow more than 2G lines, however this would be a long-term effort and I don't think cluttering our codebase with asserts would help towards that goal)